namenode/main.go

```diff
@@ -250,11 +251,101 @@ func (server *NameNode) ReportBlock(blockReport common.BlockReport, success *boo
 	return nil
 }
 
+// NameNode checks if blocks are under-replicated and schedules replication
+func (server *NameNode) replicationMonitor() {
+	slog.Info("Replication monitor setup")
+
+	for {
+		time.Sleep(common.REPLICATION_MONITOR_INTERVAL) // Check periodically
+		slog.Info("Replication monitor running")
+
+		for fileName, inode := range server.inodes {
+			inode.rwlock.RLock()
+			for blockIndex, storageInfo := range inode.storageInfo {
+				if len(storageInfo.DataNodes) < common.BLOCK_REPLICATION && storageInfo.LatestVersion > 0 {
```

Comment:
I suggest replacing this with `storageInfo.LastestVersion >= common.MIN_VALID_VERSION_NUMBER`